README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+@autoinject
+class PetsMediaType {
+  constructor(private petMediaType: PetMediaType) {}
+  mediaType: string = 'application/pets+json';
+  formatForResponse(pets: IPet[]): any {
+    return pets.map(this.petMediaType.formatForResponse);
+  }
+}
+
+// tslint:disable-next-line: max-classes-per-file
+@autoinject
+export class PetRouter {
+  constructor(private petMediaType: PetMediaType, private oldPetMediaType: OldPetMediaType, private petsMediaType: PetsMediaType){}
+  pets: IPet[] = [
+    { firstName: 'honey', lastName: 'hoguet', internalS3PathToPicture: '...' },
+    { firstName: 'poseidon', lastName: 'hoguet', internalS3PathToPicture: '...' }
+  ];
+  getRoutes(): IHTTPRoute[] {
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/')
+        .mediaType(this.petsMediaType)
+        .allowAnonymous()
+        .get(() => {
+          return this.pets;
+        }),
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .mediaType(this.oldPetMediaType)
+        .mediaType(this.petMediaType)
+        .allowAnonymous()
+        .get(({ petId }) => {
+          return this.pets[petId];
+        })
+    ];
+  }
+}
+```
+
+Once you add a custom media type, only the media types you specify will work (you've opted out of the default application/json). This is a DX issue for consumers of your API and we include a helpful response to help navigate it.
+
+```sh
+curl -H 'accept:application/json'  http://localhost:4000/pets/
+{"status":406,"message":"The requested Accept format cannot be satisfied. Try one of the supported media types: application/pets+json"}
+```
+
+> also of note that this doesn't have to be a breaking change, in fact the way we handled `/pets/:petId'` above demostrated that.
+
+> also of note is that your format methods can be async
+
+So far we have been focused on response formatting (which is 90% of the use cases in my experience, but you also can do formatting on the request.
+
+Consider this example
+
+```ts
+new RouterMetaBuilder()
+  .path('/pets/')
+  .allowAnonymous()
+  .post(({ body }) => {
+
+  })
+```
+
+What ever is posted will be in body... but maybe I want to whitelist what makes it to my handler.
+
+```ts
+@autoinject
+class postPetMediaType {
+  constructor(private petMediaType: PetMediaType) {}
+  mediaType: string = 'application/pet+json';
+  formatForResponse(pets: IPet[]): any {
+    throw new Error('not supported');
+  }
+  formatFromRequest(body: any): any {
+    const { firstName, lastName } = body;
+    return { firstName, lastName };
+  }
+}
+// and in my router...
+ new RouterMetaBuilder()
+    .path('/pets/')
+    .allowAnonymous()
+    .mediaType(this.petsMediaType)
+    .mediaType(this.postPetMediaType)
+    .post(({ model }) => {
+
+    })
+```
+
+Now, `model` is what we'd expect.
+
+> The MediaType architecture could use a v2 to embrace that it is really an adapter for the handler (which use a media type), it is not a media type!
```

Comment:
Im struggling to understand what this means.

Reply:
Create a roadmap section and move items that talk about 'next version' to there.